### ISSUE TITLE
Add a TurnRoad

### ISSUE THREAD

**Issue by ArielRotem1** [ArielRotem1/Traffic-Simulator#8]
**Is your feature request related to a problem? Please describe.**
Add a turning road

**Comment by ArielRotem1** [comment]
Yes, it would be great! I assigned the issue to myself because the issue is complex and needed number of steps to solve.

The tasks to do to complete the issue:
- Drawing the road - You need to draw white lines in the middle of the road according to the turning direction of the road.
- Need to draw the cars not according to the speed of the cars only but also according to the road type. Which means moving the draw function of the cars in the Car class to the Road class, TurnRoad class.
- Fitting the getNextRoad function to also handle TurnRoad. Copy the getNextRoad from Road class into TurnRoad class and add the necessary code to also handle TurnRoad.
- Change the moveCars of the moveCars function to also handle TurnRoad .(Copy the moveCars from Road class into TurnRoad class and add the necessary code to also handle TurnRoad. The car in the TurnRoad will drive until it pass a certain limit and the it will turn in 90 degrees to the direction it turning)(AFTER THIS COMPLETED: modify the car movement to look smooth - I will do it myself)
- Label the TurnRoad type -> {
TOP_LEFT: 1,
TOP_RIGHT: 2,
BOTTOM_LEFT: 3,
BOTTOM_RIGHT: 4
}
To know the turning direction of the road, for example: TOP_LEFT type is from right to left and from up to down and in reverse from the other way.

- **WHEN ALL TASKS COMPLETED**: Remove the option of cars to drive from one side of the canvas to the other side of the canvas.


You can choose a step to complete and it would be very helpful!
I recommend on "Drawing the road" and "Label the TurnRoad type" steps which are easy, unless you look for something more complicated : )

**Comment by ArielRotem1** [comment]
@[Volthai7us](https://github.com/Volthai7us) So do you want to contribute? if you don't want it is okay, I just want to know if I can work on it myself.

**Comment by Volthai7us** [comment]
@ArielRotem1 I cloned the project and I started but I don't want to cause any late. I could give up

**Comment by ArielRotem1** [comment]
No, it's great that you are working on it!
No rush, I just did not know you are working on it. I would love to hear
what are you currently working on? Maybe I can help if you need.
And of course that if you have any questions just let me know.

**Comment by Volthai7us** [comment]
So far I only created static enum to store TurningRoad type. Then I think skip to change cars' speed according road type.

**Comment by Volthai7us** [comment]
And I was wondering, shouldn't be more clear If we separate Class inside the scrpit.js as another js file.

**Comment by ArielRotem1** [comment]
I was going to do it, thanks for the comment.
I forgot to say but it would be best, in my opinion, that you would create
a pull request on every part of the issue you solved so it will be easier
for me to inspect it and approve it.

**Comment by Volthai7us** [comment]
I separated files. And It's working right now. I you want I could PR also these files.

**Comment by ArielRotem1** [comment]
Yes, it would be good, thanks.
